src/snowflake/cli/api/project/util.py

```diff
@@ -12,8 +12,9 @@
 SINGLE_QUOTED_STRING_LITERAL_REGEX = r"'((?:\\.|''|[^'\n])+?)'"
 
 # See https://docs.snowflake.com/en/sql-reference/identifiers-syntax for identifier syntax
-UNQUOTED_IDENTIFIER_REGEX = r"(^[a-zA-Z_])([a-zA-Z0-9_$]{0,254})"
+UNQUOTED_IDENTIFIER_REGEX = r"([a-zA-Z_])([a-zA-Z0-9_$]{0,254})"
```

Comment:
UNQUOTED_IDENTIFIER_REGEX is only ever used with re.fullmatch so this will not change existing behavior but allows it to be used in other patterns for more flexibility.